torch/_export/wrappers.py

```diff
@@ -115,3 +126,52 @@ def call(self, *args):
 
     cls.__call__ = call
     return cls
+
+
+def _mark_constructor_exportable_experimental(constructor_subclass):
+    def wrapper(*args, **kwargs):
+        # TODO container type input is not supported, will need to apply flat_apply
+        assert is_traceable_wrapper_subclass_type(type(args[0]))
+        constructor_subclass(*args, **kwargs)
+        if torch._C._is_torch_function_mode_enabled():
```

Comment:
nit: invert the condition and return early instead of all the extr indenting (this will also make it clearer that this fn is meant to short circuit and be faster when we are not actively tracing pre-dispatch)